Contents/Code/__init__.py

```
@@ -32,91 +33,180 @@ def GetApiKey():
 
 
-def HttpReq(url, authenticate=True, retry=True):
+def HttpReq(url, retry=True):
     global API_KEY
     Log("Requesting: %s" % url)
-    api_string = ''
-    if authenticate:
-        api_string = '&apikey=%s' % GetApiKey()
+
+    myheaders = {'apikey': GetApiKey()}
 
     try:
         return JSON.ObjectFromString(
-            HTTP.Request('http://%s:%s/%s%s' % (Prefs['Hostname'], Prefs['Port'], url, api_string)).content)
+            HTTP.Request('http://%s:%s/%s' % (Prefs['Hostname'], Prefs['Port'], url), headers=myheaders).content)
     except Exception, e:
         if not retry:
             raise e
 
         API_KEY = ''
-        return HttpReq(url, authenticate, False)
+        return HttpReq(url, False)
         
 
 
 class ShokoCommonAgent:
     def Search(self, results, media, lang, manual, movie):
         name = ( media.title if movie else media.show )
 
-        # http://127.0.0.1:8111/api/serie/search?query=Clannad&level=1&apikey=d422dfd2-bdc3-4219-b3bb-08b85aa65579
-
         if movie:
             if media.filename:
-                filename = os.path.basename(urllib.unquote(media.filename))
+                filename = urllib.unquote(media.filename)
+                filename = os.path.join(os.path.split(os.path.dirname(filename))[-1], os.path.basename(filename)) # Parent folder + file name
+
+                Log('Searching movie %s - %s' % (name, filename))
+
+                # Get file data using filename
+                # http://127.0.0.1:8111/api/v3/File/PathEndsWith/%5Bjoseole99%5D%20Clannad%20-%2001%20(1280x720%20Blu-ray%20H264)%20%5B8E128DF5%5D.mkv
+                file_data = HttpReq('api/v3/File/PathEndsWith/%s' % (urllib.quote(filename)))
+
+                # Take the first file. As we are searching with both parent folder and filename, there should be only one result.
+                if len(file_data) > 1:
+                    Log('File search has more than 1 result. HOW DID YOU DO IT?')
```

I would prefer that we log, and fail here, not just log and potentially give invalid results.